app/controllers/concerns/foreman/controller/smart_proxy_auth.rb

```diff
@@ -32,12 +32,22 @@ def require_smart_proxy_or_login(features = nil)
       return true
     end
 
+    # if we reach this, the requestor is a user and not a smart proxy
+    require_user_login
+  end
+
+  def require_user_login
     require_login
+    check_user_enabled
     unless User.current
       render_error 'access_denied', :status => :forbidden unless performed? && api_request?
       return false
     end
     authorize
+    verify_authenticity_token
```

Comment:
We can short-circuit here, and stop the filter process, if one method returns `false`:
``` ruby
authorize && verify_authenticity_token && set_taxonomy ...
```

Reply:
Except I don't think it's good if `session_expiry` is conditional..